inc/components/components/image/component.php

```diff
@@ -12,4 +12,28 @@
 /**
  * Register the component.
  */
-register_component_from_config( __DIR__ . '/component' );
+register_component_from_config(
+	__DIR__ . '/component',
+	[
+		'config_callback' => function( array $config ): array {
+
+			// Bail early if passed a `src` URL.
```

Comment:
If a `src` is passed, I think we should respect that URL and not try to do any additional WordPress hydration, unless we're wanting to try and sideload the image, which seems potentially undesirable.